core/src/main/scala/cromwell/util/JsonEditor.scala

```diff
@@ -85,14 +86,46 @@ object JsonEditor {
 
   def outputs(json: Json): Json = includeJson(json, NonEmptyList.of("outputs")) |> (excludeJson(_, NonEmptyList.one("calls")))
 
-  def logs(json: Json): ErrorOr[Json] = {
+  // Return the calls object with subworkflows removed.
+  private def removeSubworkflowCalls(callsObject: JsonObject): ErrorOr[Json] = {
+    def isCallSubworkflow(callArrayJson: Json): ErrorOr[Boolean] = {
+      for {
+        array <- callArrayJson.asArray map (_.validNel) getOrElse s"call array unexpectedly not an array: $callArrayJson".invalidNel
+        // All calls within this array are assumed to have the same "shape": either they are subworkflows or regular jobs.
+        // So this only looks at the first element of the call array to determine whether this member of the containing
+        // object should be filtered as a subworkflow. There should always be at least one element in this calls array.
+        callJson <- array.headOption map (_.validNel) getOrElse "call array unexpectedly empty".invalidNel
+        call <- callJson.asObject map (_.validNel) getOrElse s"Call JSON unexpectedly not an object: $callJson".invalidNel
+      } yield call.contains(subWorkflowMetadataKey)
+    }
+
+    callsObject.toList.flatTraverse[ErrorOr, (String, Json)] {
+      case (key, json) => isCallSubworkflow(json) map {
+        _.option((key, json)).toList // If the value is true return a single-element List containing the (key, json) pair,
```

Comment:
TOL: this line probably wouldn't have needed a comment if it was `isCallSubwfl => if (isCallSubwfl) List((key, json)) else List.empty` 🙂

Reply:
Yeah I agree. It turns out I had that logic backward so I had to add the parameter explicitly anyway 🙈